### PR TITLE
Feature/CON-12/Create confirmDelete dialog

### PR DESCRIPTION
**Pull request by bziondik**
Related to: https://oat-sa.atlassian.net/browse/CON-12
Related to 
https://github.com/oat-sa/extension-tao-item/pull/437
https://github.com/oat-sa/extension-tao-mediamanager/pull/236
https://github.com/oat-sa/tao-core/pull/2721

How to test:
- go to Items page
- try to delete Item
- try to delete Class

Can be tested here http://item-delete_warning.playground.kitchen.it.taocloud.org:49591/

image
image

### CODE REVIEW

**Review by habemuscode — COMMENTED** [review]
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by habemuscode — APPROVED** [review]

File: src/dialog/confirmDelete.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA ;
```
Comment (adriloma) [inline src/dialog/confirmDelete.js:16]: Better update copyright with current year.

File: src/dialog/confirmDelete.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ * @param {Object} options - Dialog options
+ * @param {Object} options.confirmationMessage - "I understand that this action is permanent." message
+ * @param {Object} options.buttons - Dialog button options
+ * @param {Object} options.buttons.labels - Dialog button labels
+ * @param {String} options.buttons.labels.delete - "Delete" button label
+ * @param {String} options.buttons.labels.cancel - "Cancel" button label
+ * @returns {dialog} - Returns the dialog instance
+ */
+export default function dialogConfirm(message, accept, refuse, options) {
+    var accepted = false;
```
Comment (adriloma) [inline src/dialog/confirmDelete.js:40]: The use of `var` here instead of `let` / `const` is intentional?

**Review by adriloma — CHANGES_REQUESTED** [review]
Made a couple of minor comments. 

Would also be valuable to add some unit tests for the new component?

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] bumping versions are applied (if applicable)

File: src/dialog/confirmDelete.js (hunk 1 above)
Comment (habemuscode) [reply src/dialog/confirmDelete.js]: done

File: src/dialog/confirmDelete.js (hunk 2 above)
Comment (habemuscode) [reply src/dialog/confirmDelete.js]: Fixed

**Review by adriloma — APPROVED** [review]

File: src/dialog/confirmDelete.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            labels: {
+                delete: __('Delete'),
+                cancel: __('Cancel')
+            }
+        },
+        confirmationMessage: __('I understand that this action is permanent.')
+    };
+    let dialogOptions;
+    let dlg;
+    options = _.defaults(options || {}, _options);
```
Comment (adriloma) [inline src/dialog/confirmDelete.js:52]: `_options` was declared as `const` and is being reassigned here

**Review by adriloma — CHANGES_REQUESTED** [review]

**Review by andrey-ponamarev — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] The feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request's target is not `master`

**Review by adriloma — APPROVED** [review]

**Pull request merged**